gc.c

```diff
@@ -11460,6 +11461,10 @@ gc_stress_get(rb_execution_context_t *ec, VALUE self)
 static void
 gc_stress_set(rb_objspace_t *objspace, VALUE flag)
 {
+    if (flag != Qtrue && flag != Qfalse) {
+        flag = rb_to_int(flag);
```

Comment:
Fixnums don't need to be marked.
```suggestion
        flag = UINT2NUM(NUM2UINT(flag));
```

Reply:
What's the point of `UINT2NUM` -> `NUM2UINT`?

Reply:
Reject non-integer objects.
This may be safer, to avoid Bignum possibility:

```C
    flag = FIX2LONG(NUM2UINT(flag) & ~(~0U << gc_stress_max));
```